gosap.go

```diff
@@ -406,3 +406,210 @@ func (s *Session) CreatePurchaseDeliveryNote(cfg Config, note PurchaseDeliveryNo
 
 	return true, nil
 }
+
+func (s *Session) GetInventoryCounting(cfg Config, id int) (*InventoryCounting, error) {
+	url := cfg.GetInventoryCountingEndpoint(id)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := s.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// Read the response body
+	content, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("could not read response body content due to %s", err)
+	}
+
+	var inventoryCounting InventoryCounting
+
+	// Unmarshal the JSON into the wrapper struct
+	if err := json.Unmarshal(content, &inventoryCounting); err != nil {
+		return nil, fmt.Errorf("could not load json response due to %s", err)
+	}
+
+	return &inventoryCounting, nil
+}
+
+func (s *Session) GetInventoryCountings(cfg Config) ([]InventoryCounting, error) {
+	url := cfg.GetInventoryCountingsEndpoint()
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := s.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// Read the response body
+	content, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("could not read response body content due to %s", err)
+	}
+
+	// Unmarshal the JSON into the wrapper struct
+	var response InventoryCountingResponse
+	if err := json.Unmarshal(content, &response); err != nil {
+		return nil, fmt.Errorf("could not load json response due to %s", err)
+	}
```

Comment:
Here we have to pull all the inventory countings in the database with a recursive  call. You can take a look of how it is done with GetDeliveryNotes.